Makefile

```diff
@@ -144,3 +144,5 @@ TAGS: $(C_SOURCE_FILES)
 	etags -o $@ $(C_SOURCE_FILES)
 GPATH GRTAGS GSYMS GTAGS: $(C_SOURCE_FILES)
 	ls $(C_SOURCE_FILES) | gtags -f - --gtagsconf .globalrc
+cscope.in.out cscope.po.out cscope.out: $(C_SOURCE_FILES)
+	cscope -bq -u -Iinclude -Ilibrary $(patsubst %,-I%,$(wildcard 3rdparty/*/include)) -Itests/include $(C_SOURCE_FILES)
```

Comment:
@gilles-peskine-arm, what do you think of indexing the `.function` test files?

Reply:
It's debatable, but I chose to include them in `C_SOURCE_FILES. The test entry points aren't actually function names (the test framework renames them to `mbedtls_test_xxx`) but it's convenient to use lookup on auxiliary functions and macros defined in a test suite header or in the template files (`main_test.function`, etc.), and they don't tend to shadow useful definitions. Likewise `C_SOURCE_FILES` includes internal definitions of sample programs.

Reply:
Null hypothesis: the default way to use make to create `NAME_OF_FILE` is `make NAME_OF_FILE`. Sure, I could add
```
.PHONY: cscope
cscope: cscope.in.out cscope.po.out cscope.out
.PHONY: global
global: GPATH GRTAGS GSYMS GTAGS
```
but what would be the point?

Reply:
No, it would be just:
cscope: $(C_SOURCE_FILES)                                                       
        cscope -bq -u -Iinclude -Ilibrary $(patsubst %,-I%,$(wildcard 3rdparty/*/include)) -Itests/include $(C_SOURCE_FILES)
What are exactly the csope files does not seem very important to me.

Reply:
But then you couldn't build a file by running `make NAME_OF_FILE`. You'd have to read the makefile to figure out what target name to use. That's bad UX.

Reply:
ok thus:
cscope cscope.out ... : $(C_SOURCE_FILES)
cscope -bq -u -Iinclude -Ilibrary $(patsubst %,-I%,$(wildcard 3rdparty/*/include)) -Itests/include
I am using cscope and when I want to rebuild the tags I run `cscope ...` and I don't really care what the output file are. With you current proposal I have to learn about them. Not a big deal I can learn that one of the output files is cscope.out but I would prefer not having to.

Reply:
I don't see the point, but I don't see the harm either, so done.

Reply:
Thanks, at least it will save me some typing.